### PR TITLE
Add online.net DNS API

### PR DESCRIPTION
**Pull request by Augustin-FL**
This Pull Request add support for the Online (https://online.net/) DNS API.

Related issue : #2093

### CODE REVIEW

**Comment by Neilpang** [comment]
please confirm whether you have checked EVERY item in this guide : https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide

Thanks.

**Comment by Augustin-FL** [comment]
@Neilpang yes, I have checked every item in this guide.

1. use cloudflare as reference : yes. OVH has also been used as reference
2. script name having the correct name: yes
3. location in `dnsapi/` : yes
4. shebang : use `sh` from env, not something else : yes
5. has `dns_myapi_<add|rm>` functions: yes
6. add function guides:
   6.1. gets fulldomain/txtvalue in dns_myapi_add(): yes
   6.2.  save credentials in _add() : yes, via _online_check_config()
   6.3. detect root zone: yes, via _get_root(). I have tested 
   `example.org`
   `subdomain.example.org`
   `subsubdomain.subdomain.example.org`
   `domainthatidontown.org`
   `invaliddomain`
   6.4. use `_post`/ `_get`, not `wget` / `curl` : yes, in _online_rest() function. 
   6.5 process strings using provided functions and some basic commands (`cut`,...), not `awk`: yes.
7. rm function runs standalone: yes, records are correctly removed.
8. support wildcard txt certs: yes, tested with `*.example.org`
9. updated docs accordingly : yes
10. creating issue related to this DNS provider : yes

File: dnsapi/dns_online.sh [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  _info "Destroying temporary version"
+  _online_destroy_zone "$_temporary_dns_version"
+
+  return 0
+}
+
+####################  Private functions below ##################################
+
+_online_check_config() {
+
```
Comment (Neilpang) [inline dnsapi/dns_online.sh]: please use `_readaccountconf_mutable` to read ONLINE_API_KEY first.

https://github.com/Neilpang/acme.sh/blob/master/dnsapi/dns_cf.sh#L17

File: dnsapi/dns_online.sh [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  if [ -z "$ONLINE_API_KEY" ]; then
+    _err "No API key specified for Online API."
+    _err "Create your key and export it as ONLINE_API_KEY"
+    return 1
+  fi
+
+  _saveaccountconf ONLINE_API_KEY "$ONLINE_API_KEY"
```
Comment (Neilpang) [inline dnsapi/dns_online.sh]: please use `_saveaccountconf_mutable`

File: dnsapi/dns_online.sh [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+_online_rest() {
+  m=$1
+  ep="$2"
+  data="$3"
+  _debug "$ep"
+  _online_url="$ONLINE_API/$ep"
+  _debug2 _online_url "$_online_url"
+  export _H1="Authorization: Bearer $ONLINE_API_KEY"
+  export _H2="X-Pretty-JSON: 1"
+  if [ "$data" ] || [ "$m" = "PATCH" ] || [ "$m" = "POST" ] || [ "$m" = "PUT" ] || [ "$m" = "DELETE" ]; then
```
Comment (Neilpang) [inline dnsapi/dns_online.sh]: why not just use `[ "$m"  != "GET" ]`

**Comment by Neilpang** [comment]
Please confirm this is ready to merge, and this is still working as expected.

Thanks.

**Comment by Augustin-FL** [comment]
I confirm that it is still working as expected (I performed tests again to make sure)

@Neilpang this PR is ready to be merged !

**Pull request merged**